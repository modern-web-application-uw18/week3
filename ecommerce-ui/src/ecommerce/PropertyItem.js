import React, { Component } from 'react';
import PropTypes from 'prop-types';

class PropertyItem extends Component {

  render() {
    return (
      <div className="row">
        <div className="col-4 items">
          <h2>{this.props.title}</h2>
          <img src={this.props.image} alt={this.props.title} />
          <p>Host: {this.props.host}, offering their: {this.props.type}</p>
          <p>Where: {this.props.city}, {this.props.country}</p>
          <h4>Cost: {'$' + this.props.cost}</h4>
          <button value={this.props.id} onClick={this.props.addToCart}
            className="cartBtn">Add to cart!</button>
        </div>
      </div>
    );
  }
}

PropertyItem.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  host: PropTypes.string,
  type: PropTypes.string,
  cost: PropTypes.number,
  city: PropTypes.string,
  country: PropTypes.string
};

export default PropertyItem;
