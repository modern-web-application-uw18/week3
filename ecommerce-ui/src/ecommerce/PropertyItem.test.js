import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import PropertyItem from './PropertyItem'

describe("PropertyItem", () => {
	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<PropertyItem />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	it ('Should render with correct info on the individual properties', () => {
		const component = TestRenderer.create(<PropertyItem />);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
