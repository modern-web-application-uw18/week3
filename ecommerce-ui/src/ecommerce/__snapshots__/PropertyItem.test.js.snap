// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyItem Should render with correct info on the individual properties 1`] = `
<div
  className="row"
>
  <div
    className="col-4 items"
  >
    <h2 />
    <img
      alt={undefined}
      src={undefined}
    />
    <p>
      Host: 
      , offering their: 
    </p>
    <p>
      Where: 
      , 
    </p>
    <h4>
      Cost: 
      $undefined
    </h4>
    <button
      className="cartBtn"
      onClick={undefined}
      value={undefined}
    >
      Add to cart!
    </button>
  </div>
</div>
`;
